src/core/InstrumentFunctions.cpp

```diff
@@ -435,26 +469,27 @@ void InstrumentFunctionArpeggio::processNote( NotePlayHandle * _n )
 		// process according to arpeggio-direction...
 		if (dir == ArpDirection::Up || dir == ArpDirection::Down)
 		{
-			cur_arp_idx = ( cur_frame / arp_frames ) % range;
+			cur_arp_idx = (cur_frame / arp_frames) % (range + sortOffset);
 		}
-		else if ((dir == ArpDirection::UpAndDown || dir == ArpDirection::DownAndUp) && range > 1)
+		else if ((dir == ArpDirection::UpAndDown || dir == ArpDirection::DownAndUp) &&
+			(range + sortOffset) > 1)
 		{
 			// imagine, we had to play the arp once up and then
 			// once down -> makes 2 * range possible notes...
 			// because we don't play the lower and upper notes
 			// twice, we have to subtract 2
-			cur_arp_idx = (cur_frame / arp_frames) % (range * 2 - (2 * static_cast<int>(m_arpRepeatsModel.value())));
+			cur_arp_idx = (cur_frame / arp_frames) % ((range + sortOffset) * 2 - (2 * static_cast<int>(m_arpRepeatsModel.value())));
 			// if greater than range, we have to play down...
 			// looks like the code for arp_dir==DOWN... :)
-			if (cur_arp_idx >= range)
+			if (cur_arp_idx >= (range + sortOffset))
 			{
-				cur_arp_idx = range - cur_arp_idx % (range - 1) - static_cast<int>(m_arpRepeatsModel.value());
+				cur_arp_idx = (range + sortOffset) - cur_arp_idx % (range + sortOffset - 1) - static_cast<int>(m_arpRepeatsModel.value());
 			}
 		}
 		else if( dir == ArpDirection::Random )
 		{
 			// just pick a random chord-index
-			cur_arp_idx = (int)( range * ( (float) rand() / (float) RAND_MAX ) );
+			cur_arp_idx = (int)((range + sortOffset) * ((float) rand() / (float) RAND_MAX));
```

Comment:
Convert to static_cast
```suggestion
			cur_arp_idx = static_cast<int>((range + sortOffset) * (static_cast<float>(rand()) / static_cast<float>(RAND_MAX)));
```